From a917148f469be6cb8a8fbae888634bf2e75ce50d Mon Sep 17 00:00:00 2001
From: Aaradhana Sahu <quic_aarasahu@quicinc.com>
Date: Fri, 17 Feb 2023 16:21:58 +0530
Subject: [PATCH] cfg80211: add support rts threshold readable command for MLO

To avoid creating rts_threshold file for MLO.

Existing rts_threshold for non MLO case
cat /sys/kernel/debug/ieee80211/phyX/rts_threshold

proposed solution for single wiphy architecture
SLO: cat /sys/kernel/debug/ieee80211/phy0/netdev:wlanX/link0/rts_threshold
MLO: cat /sys/kernel/debug/ieee80211/phy0/netdev:wlanX/linkY/rts_threshold

Signed-off-by: Aaradhana Sahu <quic_aarasahu@quicinc.com>
---
 net/wireless/debugfs.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/net/wireless/debugfs.c b/net/wireless/debugfs.c
index aab4346..701fcb6 100644
--- a/net/wireless/debugfs.c
+++ b/net/wireless/debugfs.c
@@ -102,7 +102,9 @@ void cfg80211_debugfs_rdev_add(struct cfg80211_registered_device *rdev)
 {
 	struct dentry *phyd = rdev->wiphy.debugfsdir;
 
-	DEBUGFS_ADD(rts_threshold);
+	if (!rdev->wiphy.num_hw)
+		DEBUGFS_ADD(rts_threshold);
+
 	DEBUGFS_ADD(fragmentation_threshold);
 	DEBUGFS_ADD(short_retry_limit);
 	DEBUGFS_ADD(long_retry_limit);
-- 
2.17.1

